rohmu/object_storage/config.py

```diff
@@ -76,27 +77,30 @@ def calculate_s3_chunk_size() -> int:
 
 
 class AzureObjectStorageConfig(StorageModel):
-    bucket_name: str
+    bucket_name: Optional[str]
     account_name: str
-    account_key: Optional[str] = None
-    sas_token: Optional[str] = None
+    account_key: Optional[str] = Field(None, repr=False)
+    sas_token: Optional[str] = Field(None, repr=False)
     prefix: Optional[str] = None
     azure_cloud: Optional[str] = None
     proxy_info: Optional[ProxyInfo] = None
+    storage_type: Literal[StorageDriver.azure] = StorageDriver.azure
 
 
 class GoogleObjectStorageConfig(StorageModel):
     project_id: str
-    bucket_name: str
-    credential_file: Optional[str] = None
-    credentials: Optional[Dict[str, Any]] = None
+    bucket_name: Optional[str]
+    credential_file: Optional[FilePath] = None
+    credentials: Optional[Dict[str, Any]] = Field(None, repr=False)
     proxy_info: Optional[ProxyInfo] = None
     prefix: Optional[str] = None
+    storage_type: Literal[StorageDriver.google] = StorageDriver.google
 
 
 class LocalObjectStorageConfig(StorageModel):
-    directory: str
+    directory: DirectoryPath
```

Comment:
It did break things :(
It became impossible to create that model without the validation failing because the folder does not exist at the time the configuration is created. That breaks tests and will break use cases where the configuration is prepared and (de)serialized in a different environment (chroot, container, server, etc...) or just before that directory is created.

I'll prepare a revert PR.